flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/cache/CachedDataInputStream.java

```diff
@@ -102,15 +102,17 @@ private FSDataInputStream getStream() throws IOException {
         if (isFlinkThread()) {
             cacheEntry.touch();
         }
-        FSDataInputStream stream = tryGetCacheStream();
-        if (stream != null) {
-            fileBasedCache.incHitCounter();
-            return stream;
-        }
-
-        if (streamStatus == StreamStatus.CACHED_CLOSED
-                || streamStatus == StreamStatus.CACHED_CLOSING) {
+        // Repeat until get a stream
+        while (true) {
```

Comment:
nit: this is just a suggestion - so feel free to ignore.

I would have the 
```
FSDataInputStream stream = tryGetCacheStream();
while (stream == null) {
....
}
return stream;
```
Assign originalString to stream when you want to return it.